projects/ComposeConstraintLayout/macrobenchmark-app/src/main/java/com/example/macrobenchmark/newmessage/NewMessage.kt

```diff
@@ -128,164 +130,162 @@ private fun messageMotionSceneDsl(initialState: NewMessageLayout): MotionScene {
-//            }
-//            constrain(title) {
-//                width = Dimension.fillToConstraints
-//                top.linkTo(box.top)
-//                bottom.linkTo(editClose.bottom)
-//                start.linkTo(box.start, 8.dp)
-//                end.linkTo(minIcon.start, 8.dp)
-//                customColor("content", onSurface)
-//            }
-//            constrain(content) {
-//                width = Dimension.fillToConstraints
-//                height = Dimension.fillToConstraints
-//                start.linkTo(box.start, 8.dp)
-//                end.linkTo(box.end, 8.dp)
-//                top.linkTo(editClose.bottom, 8.dp)
-//                bottom.linkTo(box.bottom, 8.dp)
-//            }
-//        }
-//        val mini = constraintSet(NewMessageLayout.Mini.name) {
-//            constrain(box) {
-//                width = Dimension.value(220.dp)
-//                height = Dimension.value(50.dp)
-//
-//                end.linkTo(parent.end, 12.dp)
-//                bottom.linkTo(parent.bottom, 12.dp)
-//
-//                customColor("background", primaryVariant)
-//            }
-//            constrain(minIcon) {
-//                width = Dimension.value(40.dp)
-//                height = Dimension.value(40.dp)
-//
-//                end.linkTo(editClose.start, 8.dp)
-//                top.linkTo(editClose.top)
-//
-//                rotationZ = 180f
-//
-//                customColor("content", onPrimary)
-//            }
-//            constrain(editClose) {
-//                width = Dimension.value(40.dp)
-//                height = Dimension.value(40.dp)
-//
-//                end.linkTo(box.end, 4.dp)
-//                top.linkTo(box.top, 4.dp)
-//                customColor("content", onPrimary)
-//            }
-//            constrain(title) {
-//                width = Dimension.fillToConstraints
-//                top.linkTo(box.top)
-//                bottom.linkTo(editClose.bottom)
-//                start.linkTo(box.start, 8.dp)
-//                end.linkTo(minIcon.start, 8.dp)
-//                customColor("content", onPrimary)
-//            }
-//            constrain(content) {
-//                width = Dimension.fillToConstraints
-//                start.linkTo(box.start, 8.dp)
-//                end.linkTo(box.end, 8.dp)
-//
-//                top.linkTo(editClose.bottom, 8.dp)
-//                bottom.linkTo(box.bottom, 8.dp)
-//
-//                visibility = Visibility.Gone
-//            }
-//        }
-//
-//        fun constraintSetFor(layoutState: NewMessageLayout) =
-//            when (layoutState) {
-//                NewMessageLayout.Full -> full
-//                NewMessageLayout.Mini -> mini
-//                NewMessageLayout.Fab -> fab
-//            }
-//        defaultTransition(
-//            from = constraintSetFor(startState),
-//            to = constraintSetFor(endState)
-//        )
-//    }
+    return MotionScene {
+        val box = createRefFor("box")
```

Comment:
It would be nice but that pattern doesn't work here since each Composable has to be paired with each reference. Eg: `Box(Modifier.layoutId("box"))`. So we need a explicit name for each of them.